lib/internal/modules/esm/module_job.js

```diff
@@ -75,11 +77,22 @@ const isCommonJSGlobalLikeNotDefinedError = (errorMessage) =>
  * @returns {void}
  */
 const explainCommonJSGlobalLikeNotDefinedError = (e, url, hasTopLevelAwait) => {
-  if (e?.name === 'ReferenceError' &&
-      isCommonJSGlobalLikeNotDefinedError(e.message)) {
+  const undefinedGlobal = getUndefinedCJSGlobalLike(e?.message);
+  if (e?.name === 'ReferenceError' && undefinedGlobal !== null) {
 
     if (hasTopLevelAwait) {
-      e.message = `Cannot determine intended module format because both require() and top-level await are present. If the code is intended to be CommonJS, wrap await in an async function. If the code is intended to be an ES module, replace require() with import.`;
+      let advice = '';
+      if (undefinedGlobal === 'require') {
+        advice = 'replace require() with import';
+      } else if (undefinedGlobal === 'module' || undefinedGlobal === 'exports') {
+        advice = 'use export instead of module.exports/exports';
+      } else if (undefinedGlobal === '__filename') {
+        advice = 'use import.meta.filename instead';
+      } else if (undefinedGlobal === '__dirname') {
+        advice = 'use import.meta.dirname instead';
+      }
```

Comment:
```suggestion
      switch (notDefinedGlobalLike) {
        case 'require':
        advice = 'replace require() with import';
        break;
        
        case 'module':
        case 'exports':
        advice = 'use export instead of module.exports/exports';
        break;

        case '__filename':
        advice = 'use import.meta.filename instead';
        break;
        
        case '__dirname':
        advice = 'use import.meta.dirname instead';
        break;
      }
```